src/Model/Aggregate.php

```diff
@@ -0,0 +1,294 @@
+
+        // this model should always be read-only
+        $this->read_only = true;
+
+        parent::__construct($baseModel->persistence, $defaults);
+
+        // always use table prefixes for this model
+        $this->persistence_data['use_table_prefixes'] = true;
+    }
+
+    /**
+     * Specify a single field or array of fields on which we will group model.
+     *
+     * @param mixed[] $aggregateExpressions Array of aggregate expressions with alias as key
+     *
+     * @return $this
+     */
+    public function groupBy(array $fields, array $aggregateExpressions = [])
+    {
+        $this->groupByFields = array_unique(array_merge($this->groupByFields, $fields));
+
+        foreach ($fields as $fieldName) {
+            $this->addField($fieldName);
+        }
+
+        foreach ($aggregateExpressions as $name => $expr) {
+            $this->aggregateExpressions[$name] = $expr;
+
+            $seed = is_array($expr) ? $expr : [$expr];
+
+            $args = [];
+            // if field originally defined in the parent model, then it can be used as part of expression
+            if ($this->baseModel->hasField($name)) {
+                $args = [$this->baseModel->getField($name)];
+            }
+
+            $seed['expr'] = $this->baseModel->expr($seed[0] ?? $seed['expr'], $args);
+
+            // now add the expressions here
+            $this->addExpression($name, $seed);
+        }
+
+        return $this;
+    }
+
+    public function getRef(string $link): Reference
+    {
+        return $this->baseModel->getRef($link);
+    }
+
+    /**
+     * Method to enable commutative usage of methods enabling both of below
+     * Resulting in Aggregate on $model.
+     *
+     * $model->groupBy(['abc'])->withAggregateField('xyz');
+     *
+     * and
+     *
+     * $model->withAggregateField('xyz')->groupBy(['abc']);
+     */
+    public function withAggregateField(string $name, $seed = []): Model
+    {
+        static::addField($name, $seed);
+
+        return $this;
+    }
+
+    /**
+     * Adds new field into model.
+     *
+     * @param array|object $seed
+     */
+    public function addField(string $name, $seed = []): Field
+    {
+        $seed = is_array($seed) ? $seed : [$seed];
+
+        if (isset($seed[0]) && $seed[0] instanceof FieldSqlExpression) {
+            return parent::addField($name, $seed[0]);
+        }
+
+        if ($seed['never_persist'] ?? false) {
+            return parent::addField($name, $seed);
+        }
+
+        if ($this->baseModel->hasField($name)) {
+            $field = clone $this->baseModel->getField($name);
+            $field->unsetOwner(); // will be new owner
+        } else {
+            $field = null;
+        }
+
+        return $field
+            ? parent::addField($name, $field)->setDefaults($seed)
+            : parent::addField($name, $seed);
+    }
+
+    public function setLimit(int $count = null, int $offset = 0)
+    {
+        $this->baseModel->setLimit($count, $offset);
```

Comment:
I think this is wrong. It should set limit on aggregated itself model not baseModel.

Reply:
Done